android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -114,7 +114,11 @@ interface Engine : WebExtensionRuntime, DataCleanable {
      * @return the newly created [EngineSession].
      */
     @MainThread
-    fun createSession(private: Boolean = false, contextId: String? = null): EngineSession
+    fun createSession(
+        private: Boolean = false,
+        enableDesktopMode: Boolean = false,
```

Comment:
General advice: This is another place to stop and consider if we should be passing this as a separate parameter. We're making a big API change here.

I see from the `GeckoEngineSession` implementation that we're using this value to set the `viewPortMode`, but this isn't the only value needed in order to change into desktop mode (we need to set the `userAgentMode` too). If we were to add the user-agent as another parameter that sounds like we are starting a pattern where we could lead to new params for each new setting in the future.

For the cases of `private` and `contextId` these are explicitly needed internally to create a GeckoSession (basically, they are special). Is enabling desktop mode one of those? I'm not sure.

We might need to consider if this change is necessary or the strategy. See the comment above where I think you've identified the problem already with a possible workaround.